### DISCUSSION TITLE
How can I skip automerging on a pinned depedency?

### DISCUSSION THREAD

**Discussion by rbel-io** [renovatebot/renovate#24404] [Request Help] [answered]
### How are you running Renovate?

Self-hosted

### If you're self-hosting Renovate, tell us what version of Renovate you run.

latest

### If you're self-hosting Renovate, select which platform you are using.

github.com

### Was this something which used to work for you, and then stopped?

I never saw this working

### Wanted end result.

Our current setup is auto-merging patches and minor dependencies updates once their is a green build.
This works fine and need no change for dependencies with a range for example.

But for pinned packages we want to create a manual PR review even though it's a minor or patch version.
Is it possible to skip pinned packages from auto-merging but still creating a PR?

If yes can you please give me an example?

### What you tried so far.

Read renovate docs, I could not find a solution.

### Relevant debug logs

<details><summary>Logs</summary>

```
Copy/paste the relevant log(s) here, between the starting and ending backticks
```

</details>

**Comment by HonkingGoose** [comment]
The `automerge` docs have a config example that uses `matchUpdateTypes` to limit the type of updates that get automerged: [^1]

> Usually you won't want to automerge all PRs, for example most people would want to leave major dependency updates to a human to review first. You could configure Renovate to automerge all but major this way:
> 
> ```json
> {
>   "packageRules": [
>     {
>       "matchUpdateTypes": ["minor", "patch", "pin", "digest"],
>       "automerge": true
>     }
>   ]
> }
> ```

I don't know which `matchUpdateTypes` items you should put in the array though. I'll let the maintainers answer that part.

Maintainers: should we have a `:automergeEverythingExceptPinned` preset?

[^1]: https://docs.renovatebot.com/configuration-options/#automerge

**Reply by secustor** [reply]
The `updateType` refers to the type of change in an update and not the current state of the value. 

If I understand the question correctly, this is rather about already pinned dependencies and not pinning PRs. 

In the first case, this is currently not possible and would require a new matcher e.g. `isPinned` for packageRules.

**Reply by HonkingGoose** [reply]
> The `updateType` refers to the type of change in an update and not the current state of the value.

Whoops, I didn't realize the difference between "type of Renovate update" and the current pinned state of the value... 🙃 

> If I understand the question correctly, this is rather about already pinned dependencies and not pinning PRs.

That's how I'm reading their question too.

> In the first case, this is currently not possible and would require a new matcher e.g. `isPinned` for packageRules.

Did you see the [`matchCurrentValue`](https://docs.renovatebot.com/configuration-options/#matchcurrentvalue) config option? Maybe that's closer to what they want? Quote from the description:

> A regex to match against the raw `currentValue` string of a dependency. Valid only within a `packageRules` object.

**Reply by secustor** [reply]
`matchCurrentValue` could be an workaround, tough is error prone as it would also include prefixes like `== ` which all have to be matched. 

Ideally we would simply expose the status which we have generated via the used versioning, tough this could be problematic as we allow users to overwrite versioning via packageRules.

**Reply by rarkins** [reply, marked as answer]
I think `matchCurrentValue` with regex is a reasonable solution here. It may not be possible to have a pattern which matches _any_ versioning scheme, but pretty easy to get it to match one type at a time (e.g. semver, pep440, etc)

**Reply by rbel-io** [reply]
So if I understand it correctly with the `matchCurrentValue` approach it means we need to define a per package rule?
That's not a manageable solution I think, since dependencies come and go and we have an uncountable amount of dependencies.

``` 
   "packageRules": [
    {
      "matchPackagePatterns": ["io.github.resilience4j"],
      "matchCurrentValue": "/^1\\./"
    }
```

**Reply by rarkins** [reply]
You do not need to do it per package, but instead per package pattern. e.g. `"matchCurrentValue": "^\\d+\\.\\d+\\.\\d+"` would match any version like `1.2.3` and not `^1.2.3`
